react-native-gutenberg-bridge/android/build.gradle

```diff
@@ -290,10 +293,16 @@ If they are changed, the isBundleUpToDate flag is switched to false. That flag i
         delete '../../node_modules'
     }
 
+    task resetExtractedRNTools(type: Delete) {
+        println "Deleting temporary folders with extracted RN tools"
+        delete nodeFolder, npmFolder, yarnFolder
+    }
+
     preBuild.dependsOn(cleanupNodeModulesFolder)
```

Comment:
Food for thought not directly related to this PR: I did notice how we're cleaning the node modules folder _after_ we generate the bundle, but we're running `resetExtractedRNTools` _before_ we create the bundle. It might help make the build a bit easier to grok if we did all the "cleaning" tasks at the same time. Could we move `cleanupNodeModulesFolder` to the beginning so that we are not left with an empty node_modules after doing a build with `BUILD_GUTENBERG_FROM_SOURCE` set to false? Put another way, by deleting the `node_modules` at the end, we have to do a `yarn install` after doing a not-from-source in order to complete a not-from source build. If we instead deleted the `node_modules` at the beginning we wouldn't have that issue (and it would also do a better job of insuring we have a clean `node_modules` folder anytime we're doing a not-from-source build).

Reply:
> Do you think we could move cleanupNodeModulesFolder to the beginning so that we are not left with an empty node_modules after doing a build with BUILD_GUTENBERG_FROM_SOURCE set to false?

When we [introduced](https://github.com/wordpress-mobile/gutenberg-mobile/commit/6d5f623d0e3b7d55c66074c611a65f087da4f8c0) the `cleanupNodeModulesFolder` task, the idea was that it had to be at the end so the (Android) artifact bundling doesn't pick up unwanted/unneeded binaries, and to "cleanup" the project folder.

I guess that leaving the node_modules folder in could be an optimization in case `yarn install` has to be run again but we need to see (1) if unwanted binaries still get picked up and (2) how often will people need it when in not-from-source mode.

You think perhaps that (1) doesn't occur anymore and (2) is already shown that we need node_modules often enough?

Reply:
> we need to see (1) if unwanted binaries still get picked up...

I had not thought about (1) at all, so we would definitely need to verify that. 

> and (2) how often will people need it when in not-from-source mode.

I believe it should speed up the build substantially any time a new js bundle was generated. Just noting that in some initial testing I did last week I think I switched out the `yarn_install` task with the plain `yarn` task to get the speed increases. I didn't have a chance to look into that, so it's just something to keep in mind as a possibility if/when we try this change.

My biggest concern is actually that the node_modules could get into bad shape and _need_ to be deleted. For that reason, if we ended up making this change, I think we should also make sure that running the `clean` task wipes out the node_modules folder. That way it will be easy for a dev to get back to a good state.

All things considered, I don't think this is the most important improvement we need in the no-jitpack builds (some change along the lines of https://github.com/wordpress-mobile/gutenberg-mobile/pull/2329 that gives us a clean submodule folder and caching is much more important imo), but if we do end up sticking with this kind of a build, I think it's definitely worth further investigation.